### PR TITLE
Product Collection: remove Beta label

### LINKED ISSUE #43397: Product Collection: Remove Beta label
1. Remove Beta label from Product Collection.
2. Prepare communication 

Note: Not yet ready to be picked up, there's set of conditions to be met before.

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR removes the Beta label from Product Collection and we make it stable.

Closes https://github.com/woocommerce/woocommerce/issues/43397

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Editor and try adding Product Collection block
2. Make sure there's `Product Collection` and not `Product Collection (Beta)`
3. Add block, choose Product Catalog
4. Make sure the block's called Product Collection in Inspector Controls

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/docs/block-references/block-references.md [hunk 1]
```diff
@@ -578,7 +578,7 @@ Payment options for your store.
 
 ## Pickup Method - woocommerce/checkout-pickup-options-block
 
-Shows local pickup options.
+Shows local pickup locations.
```
Comment (imanish003) [inline plugins/woocommerce-blocks/docs/block-references/block-references.md:581]: There are more such changes in this file. I don't think these are directly related to removing the Beta label. Are these changes intentional?

**Review by imanish003 — COMMENTED** [review]
I did smoke testing with Product Collection block and it is working as expected after removing the Beta label. I left one minor comment. Everything else looks good to me :shipit: 🎉

File: plugins/woocommerce-blocks/docs/block-references/block-references.md (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce-blocks/docs/block-references/block-references.md:581]: Changes are based on an automatic script [`build:docs:block-references`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/package.json#L56). So it gets the newest data from blocks (title, descriptions, attributes etc.).

In order to update Product Collection name in Blocks Reference file, we also get all the other updates for free.

Comment (imanish003) [reply plugins/woocommerce-blocks/docs/block-references/block-references.md:581]: Aha! Understood. I will perform smoke testing again and then approve the PR. Thanks for the explanation 🙌🏻

**Review by imanish003 — APPROVED** [review]
🚀 LGTM! Excited to see the "Beta" label removed from Product Collection block. This marks a significant milestone, reflecting the hard work and dedication put into stabilizing this. Let's roll this out! 🎉
